code/datums/materials/_material.dm

```diff
@@ -100,7 +124,23 @@ Simple datum which is instanced once per type and is used for every object of sa
 
 ///This proc is called when the material becomes the one the object is composed of the most
 /datum/material/proc/on_main_applied(atom/source, mat_amount, multiplier)
-	return
+	SHOULD_CALL_PARENT(TRUE)
+	if(!is_shiny_fishing_lure || !isitem(source))
+		return
+	RegisterSignal(source, COMSIG_ITEM_FISHING_ROD_SLOTTED, PROC_REF(on_fishing_rod_slotted))
+	RegisterSignal(source, COMSIG_ITEM_FISHING_ROD_UNSLOTTED, PROC_REF(on_fishing_rod_unslotted))
+	if(istype(source, /obj/item/fishing_rod))
+		ADD_TRAIT(source, TRAIT_ROD_ATTRACT_SHINY_LOVERS, REF(src))
+	if(istype(source.loc, /obj/item/fishing_rod))
+		on_fishing_rod_slotted(source, source.loc)
+
+/datum/material/proc/on_fishing_rod_slotted(datum/source, obj/item/fishing_rod/rod, slot)
+	SIGNAL_HANDLER
+	ADD_TRAIT(rod, TRAIT_ROD_ATTRACT_SHINY_LOVERS, REF(source))
```

Comment:
Probably shouldn't affect all mats, wood or paper aren't particularly shiny

Reply:
This proc is only registered if the material has a certain var set to TRUE.